vllm/logits_process.py

```diff
@@ -0,0 +1,86 @@
+from typing import Callable, List, Tuple, Union
+
+import torch
+
+LogitsProcessor = Union[Callable[[List[int], torch.Tensor], torch.Tensor],
+                        Callable[[List[int], List[int], torch.Tensor],
+                                 torch.Tensor]]
+"""LogitsProcessor is a function that takes a list
+of previously generated tokens, the logits tensor
+for the next token and, optionally, prompt tokens as a
+first argument, and returns a modified tensor of logits
+to sample from."""
```

Comment:
Let's import this from `vllm.sampling_params` instead of copying the definition.

Reply:
It is not copied, it is just moved from `vllm.sampling_params` to this file. Is it Ok?

It seemed more logical to me to place this `LogitsProcessor` in a file near the newly introduced `NoBadWordsLogitsProcessor` (all these "base" logits processors in one place).

Reply:
I see what you mean now. That's fine then.